pkg/cmd/bootstrap.go

```diff
@@ -168,34 +176,82 @@ func initiateInteractiveMode(io *BootstrapParameters, client *utility.Client) er
 		identifier := factory.NewDriverIdentifier(factory.Mapping(host, io.PrivateRepoDriver))
 		factory.DefaultIdentifier = identifier
 	}
-	if ui.UseInternalRegistry() {
-		io.ImageRepo = ui.EnterImageRepoInternalRegistry()
-	} else {
-		io.ImageRepo = ui.EnterImageRepoExternalRepository()
-		io.DockerConfigJSONFilename = ui.EnterDockercfg()
+	if cmd.Flag(imageRepoFlag).Changed {
+		isInternalRegistry, _, err := imagerepo.ValidateImageRepo(io.ImageRepo)
+		if !isInternalRegistry {
+			if !cmd.Flag("dockercfgjson").Changed {
+				log.Progressf("The supplied image repository has been detected as an external repository.")
+				io.DockerConfigJSONFilename = ui.EnterDockercfg()
+			}
+		}
+		if err != nil {
+			return err
+		}
+	} else if promptForAll {
+		if ui.UseInternalRegistry() {
+			io.ImageRepo = ui.EnterImageRepoInternalRegistry()
+		} else {
+			io.ImageRepo = ui.EnterImageRepoExternalRepository()
+			io.DockerConfigJSONFilename = ui.EnterDockercfg()
+		}
+	}
+	if promptForAll {
+		io.GitOpsWebhookSecret = ui.EnterGitWebhookSecret(io.GitOpsRepoURL)
+	}
+	if !cmd.Flag(serviceRepoURLFlag).Changed {
+		io.ServiceRepoURL = ui.EnterServiceRepoURL()
+	}
+	io.ServiceRepoURL = utility.AddGitSuffixIfNecessary(io.ServiceRepoURL)
+	if promptForAll {
+		io.ServiceWebhookSecret = ui.EnterGitWebhookSecret(io.ServiceRepoURL)
 	}
-	io.GitOpsWebhookSecret = ui.EnterGitWebhookSecret(io.GitOpsRepoURL)
-	io.ServiceRepoURL = utility.AddGitSuffixIfNecessary(ui.EnterServiceRepoURL())
-	io.ServiceWebhookSecret = ui.EnterGitWebhookSecret(io.ServiceRepoURL)
 	secret, err := accesstoken.GetAccessToken(io.ServiceRepoURL)
 	if err != nil && err != keyring.ErrNotFound {
 		return err
 	}
-	if secret == "" {
-		io.GitHostAccessToken = ui.EnterGitHostAccessToken(io.ServiceRepoURL)
-		io.SaveTokenKeyRing = ui.UseKeyringRingSvc()
+	if secret == "" { // We must prompt for the token
+		if !cmd.Flag(statustracker.CommitStatusTrackerSecret).Changed {
+			io.GitHostAccessToken = ui.EnterGitHostAccessToken(io.ServiceRepoURL)
+		}
+		if !cmd.Flag("save-token-keyring").Changed {
+			io.SaveTokenKeyRing = ui.UseKeyringRingSvc()
+		}
 		setAccessToken(io)
 	} else {
 		io.GitHostAccessToken = secret
 	}
-	io.CommitStatusTracker = ui.SetupCommitStatusTracker()
-	io.PushToGit = ui.SelectOptionPushToGit()
-	io.Prefix = ui.EnterPrefix()
-	io.OutputPath = ui.EnterOutputPath()
+	if !cmd.Flag("commit-status-tracker").Changed && promptForAll {
+		io.CommitStatusTracker = ui.SetupCommitStatusTracker()
+	}
+	if !cmd.Flag("push-to-git").Changed && promptForAll {
+		io.PushToGit = ui.SelectOptionPushToGit()
+	}
+	if !cmd.Flag("prefix").Changed && promptForAll {
+		io.Prefix = ui.EnterPrefix()
+	}
+	outputPathOverridden := cmd.Flag("output").Changed
+	if !outputPathOverridden {
+		// Override the default path to be ./{gitops repo name}
+		repoName, err := repoFromURL(io.GitOpsRepoURL)
+		if err != nil {
+			repoName = "gitops"
+		}
+		io.OutputPath = "./" + repoName
```

Comment:
Please don't do this, use filepath.Join instead.

Reply:
Ok.